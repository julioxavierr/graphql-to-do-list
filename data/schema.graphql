"""An object with an ID"""
interface Node {
  """The id of the object."""
  id: ID!
}

"""Information about pagination in a connection."""
type PageInfo {
  """When paginating forwards, are there more items?"""
  hasNextPage: Boolean!

  """When paginating backwards, are there more items?"""
  hasPreviousPage: Boolean!

  """When paginating backwards, the cursor to continue."""
  startCursor: String

  """When paginating forwards, the cursor to continue."""
  endCursor: String
}

"""The root of all... queries"""
type Query {
  """Fetches an object given its ID"""
  node(
    """The ID of an object"""
    id: ID
  ): Node

  """Fetches objects given their IDs"""
  nodes(
    """The IDs of objects"""
    ids: [ID!]!
  ): [Node]!
  tasks: TaskConnection
}

"""Represents a Task"""
type Task implements Node {
  """The ID of an object"""
  id: ID!

  """Description of this task"""
  description: String!

  """Wheter this task is checked/unchecked"""
  checked: Boolean!

  """date of creation"""
  createdAt: String!

  """date of last update"""
  updatedAt: String!
}

"""A connection to a list of items."""
type TaskConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [TaskEdge]
}

"""An edge in a connection."""
type TaskEdge {
  """The item at the end of the edge"""
  node: Task

  """A cursor for use in pagination"""
  cursor: String!
}
