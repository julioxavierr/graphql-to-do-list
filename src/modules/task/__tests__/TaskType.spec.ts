import { graphql } from 'graphql';
import { toGlobalId } from 'graphql-relay';

import { schema } from '../../../schema';
import {
  getContext,
  connectMongoose,
  clearDbAndRestartCounters,
  disconnectMongoose,
  createTask,
} from '../../../../test/helper';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

it('should load Task', async () => {
  const task = await createTask();

  // language=GraphQL
  const query = `
    query Q($id: ID!) {
      node(id: $id) {
        ... on Task {
          description
          checked
        }
      }
    }
  `;

  const rootValue = {};
  const context = getContext({});
  const variables = { id: toGlobalId('Task', task._id) };

  const result = await graphql(schema, query, rootValue, context, variables);
  const { node } = result.data;

  expect(node.description).toBe(task.description);
  expect(node.checked).toBe(false);
});

it('should load all tasks', async () => {
  for (let i = 0; i < 10; i++) {
    await createTask();
  }

  // language=GraphQL
  const query = `
    query Q {
      tasks {
        edges {
          node {
            description
            checked
          }
        }
      }
    }
  `;

  const rootValue = {};
  const context = getContext({});

  const result = await graphql(schema, query, rootValue, context);

  expect(result).toMatchSnapshot();
});
