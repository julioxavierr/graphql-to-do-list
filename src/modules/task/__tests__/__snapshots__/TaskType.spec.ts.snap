// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load all tasks 1`] = `
Object {
  "data": Object {
    "tasks": Object {
      "edges": Array [
        Object {
          "node": Object {
            "checked": false,
            "description": "task_10",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_9",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_8",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_7",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_6",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_5",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_4",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_3",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_2",
          },
        },
        Object {
          "node": Object {
            "checked": false,
            "description": "task_1",
          },
        },
      ],
    },
  },
}
`;
